File: nuxt_vue3_lotto/server/service/external/lottoExternalService.ts
import {type LottoDrawResult} from '@/data/model/lottoDrawResult'
import {getLastDrawnNo} from '@/utils/lottoUtils'

const LOTTO_CHECK_BASE_URL = 'https://www.dhlottery.co.kr/common.do'

export async function getLottoDrawResultFromService(drawNo?: number): Promise<LottoDrawResult | null> {
    const checkDrawNo = drawNo ? drawNo : getLastDrawnNo()
    return new Promise( async (resolve, reject) => {
        try {
            const res = await $fetch<string>(LOTTO_CHECK_BASE_URL, {
                query : {
                    method: 'getLottoNumber',
                    drwNo: checkDrawNo
                }
            })
            resolve(JSON.parse(res) as LottoDrawResult)
            //resolve(res)
        } catch (error) {
            console.log(error)
        }
        resolve(null)
    })
}

File: swift_lottoResult/lottoResult/services/external/LottoExternalService.swift
import Foundation

struct LottoExternalService {
    
    static func getLottoDrawResult(_ drwNo: Int) async throws -> LottoDrawResultModel? {
        let url = URL(string: "https://www.dhlottery.co.kr/common.do?method=getLottoNumber&drwNo=\(drwNo)")!
        let (data, _) = try await URLSession.shared.data(from: url)
        let decoder = JSONDecoder()
        do {
            let lottoDrawResult = try decoder.decode(LottoDrawResultModel.self, from: data)
            if lottoDrawResult.returnValue == LottoDrawResultModel.RETURN_VALUE_SUCCESS {
                return lottoDrawResult
            }
            return nil
        } catch {
            print("decode error \(error)")
        }
        return nil
    }
}
